library/src/main/java/com/google/android/exoplayer/metadata/Id3Parser.java

```diff
@@ -60,13 +65,46 @@ public boolean canParse(String mimeType) {
         byte[] frame = new byte[frameSize - 1];
         id3Data.readBytes(frame, 0, frameSize - 1);
 
-        int firstZeroIndex = indexOf(frame, 0, (byte) 0);
+        int firstZeroIndex = indexOfEOS(frame, 0, encoding);
         String description = new String(frame, 0, firstZeroIndex, charset);
-        int valueStartIndex = indexOfNot(frame, firstZeroIndex, (byte) 0);
-        int valueEndIndex = indexOf(frame, valueStartIndex, (byte) 0);
+        int valueStartIndex = firstZeroIndex + delimiterLength(encoding);
+        int valueEndIndex = indexOfEOS(frame, valueStartIndex, encoding);
         String value = new String(frame, valueStartIndex, valueEndIndex - valueStartIndex,
-            charset);
+                                  charset);
         metadata.put(TxxxMetadata.TYPE, new TxxxMetadata(description, value));
+      } else if (frameId0 == 'P' && frameId1 == 'R' && frameId2 == 'I' && frameId3 == 'V') {
+        // Check frame ID == PRIV
+        byte[] frame = new byte[frameSize];
+        id3Data.readBytes(frame, 0, frameSize);
+
+        int firstZeroIndex = indexOf(frame, 0, (byte) 0);
+        String owner = new String(frame, 0, firstZeroIndex, "ISO-8859-1");
+        byte[] privateData = new byte[frameSize - firstZeroIndex - 1];
+        System.arraycopy(frame, firstZeroIndex + 1, privateData, 0, frameSize - firstZeroIndex - 1);
+        metadata.put(PrivMetadata.TYPE, new PrivMetadata(owner, privateData));
+      } else if (frameId0 == 'G' && frameId1 == 'E' && frameId2 == 'O' && frameId3 == 'B') {
+        // Check frame ID == GEOB
+        int encoding = id3Data.readUnsignedByte();
+        String charset = getCharsetName(encoding);
+        byte[] frame = new byte[frameSize - 1];
+        id3Data.readBytes(frame, 0, frameSize - 1);
+
+        int firstZeroIndex = indexOf(frame, 0, (byte) 0);
+        String mimeType = new String(frame, 0, firstZeroIndex, "ISO-8859-1");
+        int filenameStartIndex = firstZeroIndex + delimiterLength(encoding);
+        int filenameEndIndex = indexOfEOS(frame, filenameStartIndex, encoding);
+        String filename = new String(frame, filenameStartIndex,
+                                     filenameEndIndex - filenameStartIndex, charset);
+        int descriptionStartIndex = filenameEndIndex + delimiterLength(encoding);
+        int descriptionEndIndex = indexOfEOS(frame, descriptionStartIndex, encoding);
+        String description = new String(frame, descriptionStartIndex,
+                                        descriptionEndIndex - descriptionStartIndex, charset);
+
+        byte[] objectData = new byte[frameSize - descriptionEndIndex - 2];
```

Comment:
Fixed. Thanks for your comment.

Andudo, just an additional comment. Following ID3 specifications, for most of ID3 tags (for example, for TXXX, PRIV or GEOB ones) you can find more than one tag of the same type in a single frame. However, given we are storing metadata in a Hashmap, and using type of the tag as the key, we are missing that possibility and we will save just the last tag of each type. 

To be honest I never see that (two ID3 tags of same type in same frame) in any HLS stream, although seems to be a possibility. A quick solution to this could be moving from a Hashmap to an ArrayList although we will lose the efficiency of looking for tags of a specific type that we have right now. Otherwise, we can set as the key a good combination of TYPE + a parameter of the ID3 tag that is unique (and different for each type of id3 tag) although this will make the Hashmap useless (difficult to know the combination of type + parameter) and at the end of the day, dev user will need to iterate Hashmap to find the metadata she is looking for that is more inefficiency that iterating over an ArrayList.

What do you think?